src/AutoRest.CSharp.V3/CodeGen/BaseCSharpWriter.cs

```diff
@@ -94,7 +91,7 @@ public CodeBlock Method(string modifiers, string type, string name, params strin
         public void DocSummary(string summary)
         {
             Line("/// <summary>");
-            string[] summaryLines = summary.Split(Environment.NewLine);
+            var summaryLines = summary.Split(Environment.NewLine);
```

Comment:
Why switch to "var"? I generally only use var when the name of the type is on the other side of the assignment. That is the coding guidelines we use on the .NET team (https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) as well on our Azure .NET guidelines.

Reply:
My programming style uses `var` in every situation it can be used. I only use explicit types when they are needed, because it makes it obvious that the type on the left-hand side is intentionally being declared. Additionally, all refactoring on every project I've worked on was easier to do if all variables are declared as `var`. Additionally, I tend to make small, well defined methods, so excessive variable declaration doesn't generally happen.

I often use variables as 'logical separation points' and less as 'reusable values'. This helps readability for humans. As for the type names on the left-hand side, as long as the variables are named appropriately, I generally think that it actually helps readability to have them all as `var`, because you just understand it to be 'variable declaration' and are not particularly worried about the 'type of the container' (the type the variable is declared as).

Here is a small bit of coding conventions for using `var`. Again, I don't follow it, but if I was to follow something, this one seems to be the most straightforward: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions#implicitly-typed-local-variables

Reply:
I recommend we use our teams coding guidelines which follows that of the .NET team which is to only use var when the type is obvious. 

I can tell you from my many years on the .NET team you read code much more then you write it and you cannot rely on tools to help you resolve types, for example doing code reviews on github, you cannot easily determine the types which makes things a lot harder to review.